src/net/conn_ssl.c

```diff
@@ -16,19 +16,28 @@ typedef struct SSLConnection
 	Connection	conn;
 	SSL_CTX    *ssl_ctx;
 	SSL		   *ssl;
+	unsigned long errcode;
 } SSLConnection;
 
+static void
+ssl_set_error(SSLConnection *conn, int err)
+{
+	conn->errcode = ERR_get_error();
+	conn->conn.err = err;
+}
+
 static int
 ssl_setup(SSLConnection *conn)
 {
 	int			ret;
 
-	conn->ssl_ctx = SSL_CTX_new(SSLv23_method());
+	conn->ssl_ctx = SSL_CTX_new(TLSv1_2_method());
```

Comment:
If I'm reading the OpenSSL docs right this will force the client to only speak TLS1.2, and and not use a future TLS version if one comes out. Is this the desired behavior?
(For the record, I believe the portable way to use TLS 1.2 or newer would be:
```C
conn->ssl_ctx = SSL_CTX_new(SSLv23_method());
SSL_CTX_set_options(conn->ssl_ctx, SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1 | SSL_OP_NO_TLSv1_1);
```

Reply:
From the docs:

```
SSLv23_method(), SSLv23_server_method(), SSLv23_client_method()

    Use of these functions is deprecated. 
```

I think the function we should use in that case is `TLS_method()`.

Reply:
It appears that `TLS_method()` is only available in OpenSSL v1.1 or greater. Should we make that the minimum version?

Reply:
I resolved this by supporting both OpenSSL v1.0.x and v1.1.x.